cmd/controller/app/options/options.go

```diff
@@ -111,6 +111,9 @@ type ControllerOptions struct {
 	// CertificateRequest -> Order. Slice of string literals that are
 	// treated as prefixes for annotation keys.
 	CopiedAnnotationPrefixes []string
+
+	//Return full Certchain including root cert for k8s CSRs
+	FullCertChain bool
```

Comment:
Would this make more sense as an option on an Issuer resource (specifically on the CA issuer) rather than as a flag? It's not ideal that we have a flag that _looks_ like it could be 'global' but is actually not IMO.

Also, if Istio is reading the CSR object, is it possible for it to read the CA from the CSR itself too and append the two instead? Not sure if we ever decided whether to use annotations for this purpose @JoshVanL?

Reply:
Do you  mean `controller.IssuerOptions`?  It seems a better place to put this option in.
Istio is reading the CSR object,especially the `status` to get the signed cert back.  Currently k8s CSR does not have a `CA` filed as our built-in certificateRequest, so this is why want to extract it from the returned cert chain directly.

Reply:
> Not sure if we ever decided whether to use annotations for this purpose @JoshVanL?

@munnerz It was experimented with and then [removed](https://github.com/jetstack/cert-manager/pull/4143) as it causes issues in the reconcile loop since it’s a separate endpoint and the CA data is non-discoverable outside the same loop. 

I wonder if cert-manger should add a feature where Issuers write CA data to a defined Secret once it knows about it. This would at least be trivial in the CA issuer case. This would also improve the use ability of [cert-manager/trust](https://github.com/cert-manager/trust) which we are experimenting with.